shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetServiceImpl.java

```diff
@@ -120,6 +117,10 @@ private Dataset updateDatasetValues(final Dataset datasetDb, final Dataset datas
 		//datasetDb.setReferencedDatasetForSuperimpositionChildrenList(dataset.getReferencedDatasetForSuperimpositionChildrenList());
 		//datasetDb.setStudyId(dataset.getStudyId());
 		datasetDb.setSubjectId(dataset.getSubjectId());
+		if (dataset.getOriginMetadata().getId().equals(dataset.getUpdatedMetadata().getId())) {
+			// Force creation of a new dataset metadata
+			dataset.getUpdatedMetadata().setId(null);
```

Comment:
This is fine but is it normal that the updated metadata has the same id at this point?

Reply:
I think that, just after the import, originMetadata and updatedMetadata are pointing to the same entity.
After this, we want to create a new one.

Reply:
But is it normal? Shouldn't updatedMetadata be at least null ?

Reply:
There are some part in the code exclusively using updatedMetadata, so when it does not exists, it points on originMetadata I guess. I agree with you that this is quite weird